packages/@lwc/synthetic-shadow/src/faux-shadow/shadow-root.ts

```diff
@@ -502,6 +509,17 @@ const NodePatchDescriptors = {
                 : this;
         },
     },
+    dispatchEvent: {
```

Comment:
Should we lift [this restriction](https://github.com/salesforce/lwc/blob/4942cc44de98cae24862df4135f443ea13959555/packages/%40lwc/engine/src/framework/restrictions.ts#L272-L286) for dispatchEvent on shadow root?

Reply:
excellent catch!

Reply:
fixed